samcli/lib/telemetry/event.py

```diff
@@ -0,0 +1,64 @@
+"""
+Represents Events and their values.
+"""
+
+from abc import ABC
+from enum import Enum
+from typing import Any, List
+
+
+class EventName(Enum):
+    """Enum for the names of available events to track."""
+
+    USED_FEATURE = "UsedFeature"
+    DEPLOY = "Deploy"
+    BUILD_RUNTIME = "BuildRuntime"
+
+
+class EventType(ABC):
+    """Abstract class for Events and the types of values they may have."""
+
+    _events = {
+        EventName.USED_FEATURE: [
+            "ESBuild",
+            "Accelerate",
+            "LocalTest",
+            "CDK",
+        ],
+        EventName.DEPLOY: [
+            "CreateChangeSetStart",
+            "CreateChangeSetInProgress",
+            "CreateChangeSetFailed",
+            "CreateChangeSetSuccess",
+        ],
+    }
+
+    @staticmethod
+    def get_accepted_values(event_name: EventName) -> List[str]:
+        """Get all acceptable values for a given Event."""
+        if event_name not in EventType._events:
+            return []
+        return EventType._events[event_name]
+
+
+class Event:
+    """Class to represent Events that occur in SAM CLI."""
+
+    event_name: EventName
+    event_value: str  # Validated by EventType.get_accepted_values to never be an arbitrary string
+
+    def __init__(self, event_name: str, event_value: str):
+        Event._verify_event_name(event_name)
+        self.event_name = EventName(event_name)
+        if event_value not in EventType.get_accepted_values(self.event_name):
+            raise KeyError(f"Event '{self.event_name.value}' does not accept value '{event_value}'.")
```

Comment:
Can you create a new exception class for this instead of using native `KeyError`? That would make debugging easier. Also, consider where this Event instance will be created, and thus where the potential places that we need to handle this exception.

Reply:
Very true! Will implement